src/sqlfluff/dialects/dialect_postgres.py

```diff
@@ -3416,3 +3417,82 @@ class TruncateStatementSegment(BaseSegment):
             optional=True,
         ),
     )
+
+
+@postgres_dialect.segment()
+class CopyStatementSegment(BaseSegment):
+    """A `COPY` statement.
+
+    As Specified in https://www.postgresql.org/docs/14/sql-copy.html
+    """
+
+    type = "copy_statement"
+
+    _table_definition = Sequence(
+        Ref("TableReferenceSegment"),
+        Bracketed(Delimited(Ref("ColumnReferenceSegment")), optional=True),
+    )
+
+    _option = match_grammar = Sequence(
+        Ref.keyword("WITH", optional=True),
+        Bracketed(
+            Delimited(
+                AnyNumberOf(
```

Comment:
Can these be duplicated? If not then we recently added `AnySetOf` which prevents that so might be a better option here?

Reply:
They can't. I've tested it with Postgres and got this error:

```
ERROR:  conflicting or redundant options
```